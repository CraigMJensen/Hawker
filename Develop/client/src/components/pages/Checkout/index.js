// import React from 'react';
// import { ExternalLink } from 'react-external-link';
// import {useStripe, useElements, PaymentElement} from '@stripe/react-stripe-js';


// const Checkout = () => {

import React, { useEffect, useState } from 'react';
import { useLazyQuery, useQuery } from '@apollo/client';
import { QUERY_CHECKOUT, QUERY_SINGLE_ITEM } from '../../../utils/queries';
import { loadStripe } from '@stripe/stripe-js';
import Confirmation from '../Confirmation';
import { ExternalLink } from 'react-external-link';

const stripePromise = loadStripe('pk_test_TYooMQauvdEDq54NiTphI7jx');

const Checkout = () => {
  let itemId = window.location.href.substring(55);
  const itemIds = useQuery(QUERY_SINGLE_ITEM, { variables: { id: itemId } });
  const [getCheckout, { data }] = useLazyQuery(QUERY_CHECKOUT);

  console.info(data)
  const details = JSON.parse(localStorage.getItem("itemData")).item


  const [modalOpen, setModalOpen] = useState(false);

  useEffect(() => {
    if (data) {
      stripePromise.then((res) => {
        res.redirectToCheckout({ sessionId: data.checkout.session });
      });
    }
  }, [data]);

  function submitCheckout() {
    const id = [];
    id.push(itemIds);
    getCheckout({
      variables: { items: id }
    });
  }

  const tax = (8.25 / 100) * details.price;
  const shipping = (5 / 100) * details.price;
  const total = details.price + tax + shipping;

  return (
    <div>
    <div>
    <h1>Purchase Confirmation</h1>
    <h3>Shipping Details</h3>
    <button className='openModalBtn'
    onClick={() => {
      setModalOpen(true);
    }}
    >
      Add
    </button>
    {modalOpen && <Confirmation setModalOpen={setModalOpen} /> }
    </div>
    <p>Item Price-   {details.price}</p>
    <p>Sales Tax-    {tax}</p>
    <p>Shipping-     {shipping}</p>
    <p>Total-     {total}</p>
    <div><img src={details.image} alt={details.title} style={{width: '10rem'}}></img></div>
    <div><button onClick={submitCheckout}><ExternalLink href="https://buy.stripe.com/test_28o9CgcqlgDD7vO7ss">
      <span>Checkout</span>
    </ExternalLink></button>
    <div>
    {/* <form onSubmit={handleSubmit}>
      <PaymentElement />
      <button onClick={!stripe}>Checkout</button>
    </form> */}
    </div>
    </div>
  </div>
  );
};

export default Checkout;
