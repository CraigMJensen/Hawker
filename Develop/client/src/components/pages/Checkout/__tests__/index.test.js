// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Checkout from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('Checkout component', () => {
  // verify the component is rendering properly
  it('Checkout component renders', () => {
    render(<Checkout />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Checkout />);
    expect(asFragment()).toMatchSnapshot();
  });
});