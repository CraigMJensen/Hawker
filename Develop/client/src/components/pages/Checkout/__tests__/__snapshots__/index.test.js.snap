// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Checkout
    </h2>
  </div>
</DocumentFragment>
`;
